.github/workflows/test.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Tests
+
+on:
+  push:
+    branches: [ master, feature/package-refactoring ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/setup.py') }}
```

Comment:
The CI installs oscovida and all of its dependencies each time it runs, this usually would mean that it has to download the wheel/source code for each of the dependencies every time it runs.

Here I use the caching feature of GitHub Workflows to cache the `~/.cache/pip` directory (meaning that after the workflow finishes, the directory is compressed and stored as an artefact), this cache is then associated with a key set to `$OS-pip-$SETUP_HASH`.

When another workflow runs if that key matches up to an available cache, then it can just re-use the files which were already downloaded, which saves a bit of time since there are quite a lot of dependencies.

As the cache key includes a hash of the `setup.py` file, if you change the setup file it won't use the cache, this is done to make sure that an old cache isn't accidentally used if the dependencies specified in the setup file have changed.